### DISCUSSION TITLE
Is the VHD a Generalized Image?

### DISCUSSION THREAD

**Discussion by Malaw80s** [actions/virtual-environments#5357] [General questions] [answered]
Hi Everyone:

I have a quick question...does the VHD that gets created from the "**GenerateResourcesAndImage**" have a generalized Linux\Windows Image? or do I need to spin up a VM from that VHD and do the generalization steps manually?

A generalized image is: https://docs.microsoft.com/en-us/azure/virtual-machines/generalize

I'm asking because I would like to like to move the Image that gets created to an [Azure compute Gallery](https://docs.microsoft.com/en-us/azure/virtual-machines/shared-image-galleries)
And then utilize that image to spin up a new VM or VMSS in Azure Portal.

Thank you

**Comment by al-cheb** [comment, marked as answer]
generalization steps manually? - Yep, that's right, Linux\Windows are already generalized and no need any additional steps.

E.g. function how we upload images to a gallery:
```
function New-ImageFromVHD
{
    # Create an image from a VHD
    $osVhdUri = "https://${imageStorageAccount}.blob.core.windows.net/vhds/${imageBlobName}"
    Write-Host "Creating an image with '${imageName}' name from '${osVhdUri}' VHD that uses '${imageStorageAccount}' storage account..."
    $imageConfig = New-AzImageConfig -Location $location
    $imageConfig = Set-AzImageOsDisk -Image $imageConfig -OsType $imageOperatingSystem -OsState Generalized -BlobUri $osVhdUri
    New-AzImage -ImageName $imageName -ResourceGroupName $imageResourceGroup -Image $imageConfig
}
```

**Reply by Malaw80s** [reply]
Thank you so much @al-cheb for your fast response, I'll try out the function that you attached and let you know how it goes.

**Reply by Malaw80s** [reply]
Hi @al-cheb,
I was able to create an image from the VHD utilizing the function above, then I used the "Clone to a VM Image" option in the Azure portal to clone it to my Azure compute gallery.

Thanks again for your help :)
